src/gui/qgshttpheaderwidget.cpp

```diff
@@ -0,0 +1,129 @@
+/***************************************************************************
+                       qgshttpheaderswidget.cpp
+  This class implements simple UI for http header.
+
+                              -------------------
+          begin                : 2021-09-09
+          copyright            : (C) 2021 B. De Mezzo
+          email                : benoit dot de dot mezzo at oslandia dot com
+
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgshttpheaderwidget.h"
+#include "ui_qgshttpheaderwidget.h"
+#include "qgsapplication.h"
+
+
+QgsHttpHeaderWidget::QgsHttpHeaderWidget( QWidget *parent ) :
+  QWidget( parent )
+{
+  setupUi( this );
+  btnAddQueryPair->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/symbologyAdd.svg" ) ) );
+  btnRemoveQueryPair->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/symbologyRemove.svg" ) ) );
+  grpbxAdvanced->setCollapsed( true );
+
+  setupConnections();
+}
+
+QgsHttpHeaderWidget::~QgsHttpHeaderWidget()
+{
+  // nope
+}
+
+void QgsHttpHeaderWidget::setupConnections()
+{
+  // Action and interaction connections
+  connect( btnAddQueryPair, &QToolButton::clicked, this, &QgsHttpHeaderWidget::addQueryPair );
+  connect( btnRemoveQueryPair, &QToolButton::clicked, this, &QgsHttpHeaderWidget::removeQueryPair );
+}
+
+void QgsHttpHeaderWidget::addQueryPairRow( const QString &key, const QString &val )
+{
+  const int rowCnt = tblwdgQueryPairs->rowCount();
+  tblwdgQueryPairs->insertRow( rowCnt );
+
+  const Qt::ItemFlags itmFlags = Qt::ItemIsEnabled | Qt::ItemIsSelectable
+                                 | Qt::ItemIsEditable | Qt::ItemIsDropEnabled;
+
+  QTableWidgetItem *keyItm = new QTableWidgetItem( key );
```

Comment:
```suggestion
  QTableWidgetItem *keyItem = new QTableWidgetItem( key );
```

None of us will get RSI from the extra keystrokes, and it's helping out non-English native speakers :laughing:

Reply:
Argg.. I did not get it :sweat_smile:... what do you mean by RSI?

Reply:
"repetitive strain injury" :laughing:

Reply:
okkkk! Indeed! And I have already made the changes!